### PR TITLE
Use :async to company-phpactor, Version 2

### PR DESCRIPTION
**Pull request by zonuexe**
refs https://github.com/emacs-php/phpactor.el/pull/129

This PR was based on #129 and made function calls asynchronous using [`async`](https://github.com/jwiegley/emacs-async) package. (Thanks @nanasess)

### CODE REVIEW

**Comment by nanasess** [comment]
@zonuexe It worked so well! Thanks 👍

File: phpactor.el [hunk 1]
```diff
@@ -150,16 +151,15 @@ have to ensure a compatible version of phpactor is used."
...
+                             (file-exists-p (expand-file-name "composer.json" phpactor--lisp-directory)))
                         phpactor--lisp-directory
                       phpactor--remote-composer-file-url-dir)))
     (unless (file-directory-p phpactor-install-directory)
       (make-directory phpactor-install-directory))
     ;; Create .gitignore to prevent unnecessary files from being copied to GitHub
     (unless (file-exists-p (expand-file-name ".gitignore" phpactor-install-directory))
       (f-write-text "*\n" 'utf-8 (expand-file-name ".gitignore" phpactor-install-directory)))
-    (cl-loop for file in '("composer.json" "composer.lock")
+    (cl-loop for file in '("composer.json")
```
Comment (kermorgant) [inline phpactor.el]: This one leaves me doubtful. Why remove composer.lock and why do it here ?

**Comment by kermorgant** [comment]
Hi @zonuexe @nanasess 
good to see this work, thanks :-)
Not sure I can provide any insight about the code but I will try it tomorrow.

**Comment by kermorgant** [comment]
just tested, seems to work well.

On code level, I will trust you as I'm not used to async and a bit short of time to dive into it :-)

Only thing that worries me is that line with composer.lock being removed : as far as I understand, its purpose is to get identical installations of dependencies for everyone, and that sounds to me as a good thing, so what's the reason to remove it ?

**Comment by zonuexe** [comment]
`composer.lock` is essentially unnecessary if `composer.json` is properly described. There is a trouble caused by this file being present. However, since it was found that simply erasing would cause another problem for existing users, we will deal with it again in another PR.

**Pull request merged**